eth2-core test_slash_validator[100] failure on master

### What is wrong?

eth2-core failure on master, got it on this run: https://circleci.com/gh/ethereum/trinity/71305?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

In #857 

```
__________________________ test_slash_validator[100] ___________________________
[gw0] linux -- Python 3.6.9 /home/circleci/repo/.tox/py36-eth2-core/bin/python

genesis_state = <BeaconState #0 aead14e1>
config = Eth2Config(SHARD_COUNT=1024, TARGET_COMMITTEE_SIZE=128, MAX_VALIDATORS_PER_COMMITTEE=4096, MIN_PER_EPOCH_CHURN_LIMIT=4...=16, MAX_VOLUNTARY_EXITS=16, MAX_TRANSFERS=0, DEPOSIT_CONTRACT_ADDRESS=b'\x124Vx\x90\x124Vx\x90\x124Vx\x90\x124Vx\x90')

    @pytest.mark.parametrize(
        (
            'validator_count'
        ),
        [
            (100),
        ]
    )
    def test_slash_validator(genesis_state,
                             config):
        some_epoch = (
            config.GENESIS_EPOCH + random.randrange(1, 2**32) + config.EPOCHS_PER_SLASHINGS_VECTOR
        )
        earliest_slashable_epoch = some_epoch - config.EPOCHS_PER_SLASHINGS_VECTOR
        slashable_range = range(earliest_slashable_epoch, some_epoch)
        sampling_quotient = 4
    
        state = genesis_state.copy(
            slot=compute_start_slot_of_epoch(earliest_slashable_epoch, config.SLOTS_PER_EPOCH),
        )
        validator_count_to_slash = len(state.validators) // sampling_quotient
        assert validator_count_to_slash > 1
        validator_indices_to_slash = random.sample(
            range(len(state.validators)),
            validator_count_to_slash,
        )
        # ensure case w/ one slashing in an epoch
        # by ignoring the first
        set_of_colluding_validators = validator_indices_to_slash[1:]
        # simulate multiple slashings in an epoch
        validators_grouped_by_coalition = groupby(
            lambda index: index % sampling_quotient,
            set_of_colluding_validators,
        )
        coalition_count = len(validators_grouped_by_coalition)
        slashings = {
            epoch: coalition
            for coalition, epoch in zip(
                validators_grouped_by_coalition.values(),
                random.sample(slashable_range, coalition_count),
            )
        }
        another_slashing_epoch = first(random.sample(slashable_range, 1))
        while another_slashing_epoch in slashings:
            another_slashing_epoch += 1
        slashings[another_slashing_epoch] = (validator_indices_to_slash[0],)
    
        expected_slashings = {}
        expected_individual_penalties = {}
        for epoch, coalition in slashings.items():
            for index in coalition:
                validator = state.validators[index]
                assert validator.is_active(earliest_slashable_epoch)
                assert validator.exit_epoch == FAR_FUTURE_EPOCH
                assert validator.withdrawable_epoch == FAR_FUTURE_EPOCH
    
                expected_slashings = update_in(
                    expected_slashings,
                    [epoch],
                    lambda balance: balance + state.validators[index].effective_balance,
                    default=0,
                )
                expected_individual_penalties = update_in(
                    expected_individual_penalties,
                    [index],
                    lambda penalty: (
                        penalty + (
                            state.validators[index].effective_balance //
                            config.MIN_SLASHING_PENALTY_QUOTIENT
                        )
                    ),
                    default=0,
                )
    
        # emulate slashings across the current slashable range
        expected_proposer_rewards = {}
        for epoch, coalition in slashings.items():
            state = state.copy(
                slot=compute_start_slot_of_epoch(epoch, config.SLOTS_PER_EPOCH)
            )
    
            expected_total_slashed_balance = expected_slashings[epoch]
            proposer_index = get_beacon_proposer_index(state, CommitteeConfig(config))
    
            expected_proposer_rewards = update_in(
                expected_proposer_rewards,
                [proposer_index],
                lambda reward: reward + (
                    expected_total_slashed_balance // config.WHISTLEBLOWER_REWARD_QUOTIENT
                ),
                default=0,
            )
            for index in coalition:
                state = slash_validator(state, index, config)
    
        state = state.copy(
            slot=compute_start_slot_of_epoch(some_epoch, config.SLOTS_PER_EPOCH)
        )
        # verify result
        for epoch, coalition in slashings.items():
            for index in coalition:
                validator = state.validators[index]
                assert validator.exit_epoch != FAR_FUTURE_EPOCH
                assert validator.slashed
>               assert validator.withdrawable_epoch == (
                    epoch + config.EPOCHS_PER_SLASHINGS_VECTOR
                )
E               AssertionError: assert 4049502478 == (4049494146 + 8192)
E                +  where 4049502478 = <Validator: [b'\x91l\xfe \x93$.\xc9=\xba4}\x1a\xd2\xd9\x0e=\x0c(\x8f\x95\xe3\xc7\xe4\x87\xf3\xaa\x10\x91_\xd3\x8e\xaaf...x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 32000000000, True, 0, 0, 4049502222, 4049502478]>.withdrawable_epoch
E                +  and   8192 = Eth2Config(SHARD_COUNT=1024, TARGET_COMMITTEE_SIZE=128, MAX_VALIDATORS_PER_COMMITTEE=4096, MIN_PER_EPOCH_CHURN_LIMIT=4...=16, MAX_VOLUNTARY_EXITS=16, MAX_TRANSFERS=0, DEPOSIT_CONTRACT_ADDRESS=b'\x124Vx\x90\x124Vx\x90\x124Vx\x90\x124Vx\x90').EPOCHS_PER_SLASHINGS_VECTOR

another_slashing_epoch = 4049496995
coalition  = [24, 36, 0, 72, 48, 28, ...]
coalition_count = 4
config     = Eth2Config(SHARD_COUNT=1024, TARGET_COMMITTEE_SIZE=128, MAX_VALIDATORS_PER_COMMITTEE=4096, MIN_PER_EPOCH_CHURN_LIMIT=4...=16, MAX_VOLUNTARY_EXITS=16, MAX_TRANSFERS=0, DEPOSIT_CONTRACT_ADDRESS=b'\x124Vx\x90\x124Vx\x90\x124Vx\x90\x124Vx\x90')
earliest_slashable_epoch = 4049494114
epoch      = 4049494146
expected_individual_penalties = {0: 1000000000, 9: 1000000000, 11: 1000000000, 17: 1000000000, ...}
expected_proposer_rewards = {11: 312500000, 45: 312500000, 48: 375000000, 60: 62500000, ...}
expected_slashings = {4049494146: 256000000000, 4049496995: 32000000000, 4049499936: 160000000000, 4049500035: 160000000000, ...}
expected_total_slashed_balance = 32000000000
genesis_state = <BeaconState #0 aead14e1>
index      = 24
proposer_index = 60
sampling_quotient = 4
set_of_colluding_validators = [42, 31, 24, 36, 41, 0, ...]
slashable_range = range(4049494114, 4049502306)
slashed_balance = 192000000000
slashed_epoch_index = 8
slashings  = {4049494146: [24, 36, 0, 72, 48, 28, ...], 4049496995: (34,), 4049499936: [42, 70, 58, 90, 18], 4049500035: [41, 69, 17, 81, 9], ...}
some_epoch = 4049502306
state      = <BeaconState #259168147584 51284e5b>
validator  = <Validator: [b'\x91l\xfe \x93$.\xc9=\xba4}\x1a\xd2\xd9\x0e=\x0c(\x8f\x95\xe3\xc7\xe4\x87\xf3\xaa\x10\x91_\xd3\x8e\xaaf...x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 32000000000, True, 0, 0, 4049502222, 4049502478]>
validator_count_to_slash = 25
validator_indices_to_slash = [34, 42, 31, 24, 36, 41, ...]
validators_grouped_by_coalition = {0: [24, 36, 0, 72, 48, 28, ...], 1: [41, 69, 17, 81, 9], 2: [42, 70, 58, 90, 18], 3: [31, 11, 79, 27, 67, 55]}

tests/eth2/core/beacon/test_validator_status_helpers.py:367: AssertionError
```

### How can it be fixed

:man_shrugging:

--- a/tests/eth2/core/beacon/test_validator_status_helpers.py
+++ b/tests/eth2/core/beacon/test_validator_status_helpers.py
@@ -364,8 +364,9 @@ def test_slash_validator(genesis_state,
             validator = state.validators[index]
             assert validator.exit_epoch != FAR_FUTURE_EPOCH
             assert validator.slashed
-            assert validator.withdrawable_epoch == (
-                epoch + config.EPOCHS_PER_SLASHINGS_VECTOR
+            assert validator.withdrawable_epoch == max(
+                validator.exit_epoch + config.MIN_VALIDATOR_WITHDRAWABILITY_DELAY,
+                epoch + config.EPOCHS_PER_SLASHINGS_VECTOR,
             )
 
             slashed_epoch_index = epoch % config.EPOCHS_PER_SLASHINGS_VECTOR

